backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt

```diff
@@ -123,6 +123,10 @@ private fun isExportedFunction(descriptor: FunctionDescriptor): Boolean {
     descriptor.allParameters.forEach {
         if (it.type.isGeneric()) return false
     }
+    descriptor.typeParameters.forEach{
```

Comment:
Maybe we can just filter method with any type parameters and replace check above instead?

Reply:
sounds good. could you please take a look in next commit?